feat: :sparkles: added `is_mod_active` to `ModLoaderMod`

KANAjetzt opened the pull request:
Added  `static func is_mod_active(mod_id: String) -> bool:` to `ModLoaderMod`.

closes #355 (maybe we want to keep it open for 3.5?)

KANAjetzt committed:
feat: :sparkles: added `is_mod_active` to `ModLoaderMod`

Qubus0 reviewed (approved):
can technically be shortened to a return, but i also like being explicit sometimes

KANAjetzt committed:
refactor: :recycle: shortened to a single return

also checks `mod_data` directly now to avoid logging an error if the `mod_id` is not present.
